ls1intum/Artemis
# [Feature] Highlight Changed Files for the Assessment of Programming Exercises

### Checklist

### Motivation and Context


Currently, the tutor or instructors who assess students submissions can not directly see which files were changed or newly added during the students participation on a programming exercise. This PR introduces a highlighting, so that the tutor/instructor directly know which files they need to assess and which not (as those have not been changed).

### Description

- Added new endpoint to return if a file was changed/is new or not
- Calling the new endpoint when tree structure for file browser is loaded
- Made sure this is only for manual assessment of programming exercises

### Steps for Testing


1. Log in to Artemis
2. You can create a programming exercise or use a existing one
3. Open the student code editor or the instructor code editor ("Edit in editor")
4. Check that the files are not highlighted and check in network tab that `files-change` REST call is not called
5. Go to Assessment Dashboard
6. Open/Start Assessment
7. Check that the files that were changed/are new are highlighted and check in network tab that `files-change` was called (See screenshots below)

### Test Coverage




RepositoryService.java:  86%
Coverage of RepositoryProgrammingExerciseParticipationResource is ~70%. Did not decrease through this PR as introduced method has coverage of 100%. Will be increased in a follow up.
![image](https://user-images.githubusercontent.com/41108487/100336367-60e41800-2fd6-11eb-9307-780d8f3cc212.png)

### Screenshots



When assessing programming exercises, see that the call `files-change` is called:
<img width="1319" alt="image" src="https://user-images.githubusercontent.com/41108487/100326473-4f950e80-2fca-11eb-93b5-9fc2a2c77240.png">

When accessing as student, see that the call `files-change` is not called
<img width="1471" alt="image" src="https://user-images.githubusercontent.com/41108487/100326625-7b17f900-2fca-11eb-935d-82a5ac00fdbe.png">
